javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -473,6 +473,60 @@ private Array(Object[] delegate, int startIndex, int length) {
         return ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
     }
 
+    /**
+     * Creates a Array from a seed value and a function.
+     * The function takes the seed at first.
+     * The function should return {@code None} when it's
+     * done generating the Array, otherwise {@code Some} {@code Tuple}
+     * of the element for the next call and the value to add to the
+     * resulting Array.
+     * <p>
+     * Example:
+     * <pre>
+     * <code>
+     * Array.unfoldRight(10, x -&gt; x == 0
+     *             ? Option.none()
+     *             : Option.of(new Tuple2&lt;gt;(x, x-1)));
+     * // Array(10, 9, 8, 7, 6, 5, 4, 3, 2, 1))
+     * </code>
+     * </pre>
+     *
+     * @param seed the start value for the iteration
+     * @param f    the function to get the next step of the iteration
+     * @return an Array with the values built up by the iteration
+     * @throws IllegalArgumentException if {@code f} is null
+     */
+    static <T,U> Array<U> unfoldRight(T seed, Function1<T,Option<Tuple2<U,T>>> f) {
```

Comment:
We should use `Function` instead of `Function1` for method argument types (because `Function1 extends Function`).

Also I think it could make sense to widen the bounds:

```java
static <T,U> Array<U> unfoldRight(T seed, Function<? super T, Option<Tuple2<? extends U, ? extends T>>> f)
```

I think we do not need `? extends Option` in that signature because the Option API always returns Option.